circle.yml

```diff
@@ -1,12 +1,21 @@
 machine:
   node:
-    version: 4.3.1
+    version: 4.3.0
   environment:
+    YARN_VERSION: 0.17.8
+    PATH: "${PATH}:${HOME}/.yarn/bin"
     ENVIRONMENT_NAME: staging
 
 dependencies:
+  pre:
+    - |
+      if [[ ! -e ~/.yarn/bin/yarn || $(yarn --version) != "${YARN_VERSION}" ]]; then
+        curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version $YARN_VERSION
+      fi
```

Comment:
From reading their docs I don't think it is. This should cache yarn's own directory (`~/.yarn~), and the package directory (`~/.yarn-cache`).

From looking through the list of preinstallations, node v4.2.0 should be preinstalled, or something like it. CircleCI has:
`circleci-nodejs-4.1.21` and `circleci-nodejs-4.2.61` listed. I can't find exactly what these are, but they look like mappings from the preinstalled major versions to minor versions. The build delay we experienced might be due to some error with these.